canonicaljson.py

```diff
@@ -132,7 +152,7 @@ def iterencode_pretty_printed_json(json_object):
     #
     # Note that it seems performance is on par or better using json from the
     # standard library as of Python 3.7.
-    import simplejson as json
+    import simplejson as json  # type: ignore[no-redef]
```

Comment:
Arg, I wish we could figure out a way to kill this, but I don't think we can due to old Synapse versions which unconditionally install it...

Reply:
Ahh, I hadn't realised that Synapse imports and uses `set_json_library`.

IMO the dependencies that we control (canonicaljson, signedjson, matrix-common?) should be specified in Synapse's metadata with semver bounds to avoid this sort of thing.

Reply:
> IMO the dependencies that we control (canonicaljson, signedjson, matrix-common?) should be specified in Synapse's metadata with semver bounds to avoid this sort of thing.

Absolutely, but we didn't set an upper bound previously and there isn't really a way to go back and fix that (unless maybe once the Python's from those versions are unsupported?) 🤷